stglib/sg/wvsnc2waves.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from ..core import qaqc, utils, waves
+
+
+def nc_to_waves(nc_filename):
+    """
+    Process burst data to wave statistics
+    """
+
+    ds = xr.open_dataset(nc_filename)
+
+    # Check to see if need to make smaller wave bursts from really long wave bursts
+    if "calculated_wave_interval" in ds.attrs:
+        # Divide large burst into smaller bursts at specified calculated_wave_interval
+        ds = make_wave_bursts(ds)
+
+    spec = waves.make_waves_ds(ds)
+
+    for k in ["wp_peak", "wh_4061", "wp_4060", "pspec"]:
+        ds[k] = spec[k]
+
+    ds = utils.create_water_level_var(ds)
+    ds = utils.create_water_depth_var(ds)
+
+    # Drop unneeded variables/attributes
+    for k in ["P_1", "P_1ac", "sample", "T_28"]:
+        if k in ds:
+            ds = ds.drop_vars(k)
+
+    ds = qaqc.drop_vars(ds)
+
+    ds = utils.trim_max_wp(ds)
+
+    ds = utils.trim_min_wh(ds)
+
+    ds = utils.trim_max_wh(ds)
+
+    ds = utils.trim_wp_ratio(ds)
+
+    # Add attrs
+    ds = utils.ds_add_wave_attrs(ds)
+
+    # assign min/max (need to do this after trimming):
+    ds = utils.add_min_max(ds)
+
+    nc_filename = ds.attrs["filename"] + "w-a.nc"
+
+    ds.to_netcdf(nc_filename, unlimited_dims=["time"])
+    utils.check_compliance(nc_filename, conventions=ds.attrs["Conventions"])
+    print("Done writing netCDF file", nc_filename)
+
+    return ds
+
+
+def make_wave_bursts(ds):
+    ds.attrs["samples_per_burst"] = int(
+        ds.attrs["calculated_wave_interval"] / ds.attrs["sample_interval"]
+    )
+
+    # Calculate how many rows to subdivide each burst
+    rows = float(ds.attrs["BurstDuration"]) / float(
+        ds.attrs["calculated_wave_interval"]
+    )
+
+    # Define time interval
+    delta_t = int(ds.attrs["calculated_wave_interval"])
+    delta_t = str(delta_t) + "s"
+
+    timestamp = []
+    for t in range(0, ds["time"].size):
```

Comment:
This type of looping is often not necessary with coordinates associated with data in an xarray Dataset. Let's talk about whether there is a more straightforward way to do this.

Reply:
I had to do it with a loop because the timestamps are uneven.  I originally tried to do it with a one-liner like this: 
new_date_rng = pd.date_range(start=ds.time[0].values, end=last_time, freq=delta_t).
But it produced an even timestamp, and Steve said that was wrong because the instrument has some downtime between bursts, so the timestamps need to be uneven.